index.js

```diff
@@ -374,3 +373,60 @@ if (submitBtns) {
     });
   });
 }
+// validate contact form
+// Get the form and form fields
+const submitContactForm = document.querySelector('#submit-contact-form');
+const contactForm = document.querySelector('form');
+const nameField = document.querySelector('#name');
+const emailField = document.querySelector('#email');
+const messageField = document.querySelector('#message');
+const successMessage = document.querySelector('.success-message');
+
+// Add a submit event listener to the form
+submitContactForm.addEventListener('click', (event) => {
+  // Prevent the form from submitting
+  event.preventDefault();
+
+  // Validate the name field
+  if (nameField.value === '') {
+    successMessage.innerHTML = 'Your form was not sent! <br /> Please enter your name';
+    nameField.focus();
+    return;
+  }
+
+  // Validate the email field
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  if (!emailRegex.test(emailField.value)) {
+    successMessage.innerHTML = 'Your form was not sent! <br /> please enter a valid email address';
+    emailField.focus();
+    return;
+  }
+  if (emailField.value !== emailField.value.toLowerCase()) {
+    successMessage.innerHTML = 'Your form was not sent! <br/> Please enter your email in lowercase';
+    emailField.focus();
+    return;
+  }
+
+  // Validate the message field
+  if (messageField.value === '') {
+    successMessage.innerHTML = 'Your form was not sent! <br /> Please enter a message';
+    messageField.focus();
+    return;
+  }
+
+  // If all fields are valid, submit the form
+  contactForm.submit();
+});
+
+window.addEventListener('load', () => {
+  const storedFormData = localStorage.getItem('formData');
+  if (storedFormData) {
+    // Parse stored form data into an object
+    const parsedFormData = JSON.parse(storedFormData);
+
+    // Populate form fields with stored data
+    nameField.value = parsedFormData.name;
+    emailField.value = parsedFormData.email;
+    messageField.value = parsedFormData.message;
+  }
+});
```

Comment:
- [Required] I think it would be good if you used `localStorage.setItem` to write the values of your `input` fields to the `local storage` currently it appears that the values of the form inputs are not being written to the local storage. In order to save to the local storage use a data object, and turn it into a string using `JSON.stringify` this will make it easy when you want to read the same data back using `localStorage.getItem`.
```JavaScript
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};

localStorage.setItem('formData', JSON.stringify(formData));`
```